middleware/afc_status.py

```
@@ -0,0 +1,209 @@
+                continue
+
+            spool_id = lane_data.get("spool_id")
+            status = lane_data.get("status")
+
+            with app_state.state_lock:
+                is_locked = app_state.lane_locks.get(lane_name, False)
+                app_state.lane_statuses[lane_name] = status
+
+                if spool_id is not None:
+                    if not is_locked:
+                        logger.info(f"AFC Sync: {lane_name} has spool {spool_id}, locking")
```

Ruff (0.15.6)

[warning] Logging statement uses f-string

(G004)